### DISCUSSION TITLE
Thoughts on "Was this page helpful?"

### DISCUSSION THREAD

**Discussion by wilhelmer** [squidfunk/mkdocs-material#3166] [Ideas]
Quick thoughts:

1. The UI could use some more love, the left-aligned title with the centered buttons looks a little weird, and maybe add a visual separator between content and the feedback section.
2. Using the page title instead of the document path as the event label looks a lot cleaner in the report and helps avoiding weird entries through, e.g., local tracking (file:// protocol). However, I understand that the page title not being unique might be a problem. Maybe add both? Event label = page title, event action = path?
3. Similarly, I'd prefer the page title over path for the `{}` placeholder. "Feedback for topic 'My Topic'" is much more readable than "Feedback for topic '/my/project/dir/misc/docs/my-topic.html'"

**Comment by squidfunk** [comment]
Thanks for the feedback!

> The UI could use some more love, the left-aligned title with the centered buttons looks a little weird, and maybe add a visual separator between content and the feedback section.

What do you mean by left-aligned title? The title is centered. Furthermore, adding a separator is trivial with CSS – just add a `border-top` which fits your taste. I didn't want to clutter up the UI, but I understand that some users might want a separation.

<img width="713" alt="Bildschirmfoto 2021-11-01 um 13 49 33" src="https://user-images.githubusercontent.com/932156/139674165-b73aee94-4e73-4773-9792-40fc0cbf2d66.png">

> Using the page title instead of the document path as the event label looks a lot cleaner in the report and helps avoiding weird entries through, e.g., local tracking (file:// protocol). However, I understand that the page title not being unique might be a problem. Maybe add both? Event label = page title, event action = path?

You can only provide one label, that's a limitation of Google Analytics, and although other analytics solutions may support more parameters, we have to settle for the lowest common denominator when implementing such a feature. It can be overridden by patching the `.../analytics/google.html` partial. Additionally, URLs provide a 1:1 mapping to the underlying file structure, which reduces cognitive load when trying to find the affected page in order to improve it.

> Similarly, I'd prefer the page title over path for the {} placeholder. "Feedback for topic 'My Topic'" is much more readable than "Feedback for topic '/my/project/dir/misc/docs/my-topic.html'"

That should be possible. However, order would be important, as we're currently leveraging Python's `format` function, but that shouldn't be a problem as we're using query parameters anyway.

**Reply by wilhelmer** [reply]
1. Okay, something wrong in my project then, it's left-aligned for me. I'll check that (and also fiddle around with our CSS).
2. Yes, I already customized it for us so that it sends the page title. Note that you can provide a label and an action for each event, so I think it would be possible to send both. See https://support.google.com/analytics/answer/1033068

**Reply by squidfunk** [reply]
>  Note that you can provide a label and an action for each event, so I think it would be possible to send both.

Interesting, so we could use `action` for the URL and `label` for the `title`. I'll play around with it!

**Reply by wilhelmer** [reply]
@squidfunk The title is centered in Chrome, but not in Firefox!

![image](https://user-images.githubusercontent.com/2234833/139677102-55e671ba-37ee-483f-96c8-0f759fba3f27.png)

**Reply by wilhelmer** [reply]
https://stackoverflow.com/questions/5881440/fieldset-legend-text-align-right-bug-in-firefox

**Reply by squidfunk** [reply]
Thanks for reporting! Fixed in [`53ac0451e`](https://github.com/squidfunk/mkdocs-material-insiders/commit/53ac0451e06ff3dd54460682a7143e22fb87769e).

**Reply by wilhelmer** [reply]
> > Similarly, I'd prefer the page title over path for the {} placeholder. "Feedback for topic 'My Topic'" is much more readable than "Feedback for topic '/my/project/dir/misc/docs/my-topic.html'"
> 
> That should be possible. However, order would be important, as we're currently leveraging Python's `format` function, but that shouldn't be a problem as we're using query parameters anyway.

It would be great if you could change that. Or maybe offer two placeholders: `{title}` and `{path}`.

That's the only thing I can't customize myself, and we're planning to go live with this feature in about two weeks.

Thanks!

**Reply by wilhelmer** [reply]
@squidfunk Is that on your list or should i make a PR for that?

**Reply by squidfunk** [reply]
You can create an issue for this. I'll have to check how we can integrate this functionality because I really don't want to introduce named placeholders, so we can use Python's `format` functionality, as already explained in my original comment.

**Comment by wilhelmer** [comment]
Another one: You probably don't want the feedback buttons on the front page. Or on other "unrelated" pages like Legal Info. A way to disable the button on specific pages would be nice.

**Reply by squidfunk** [reply]
We can add a `hide.feedback` setting to disable it via front matter 😊

**Reply by wilhelmer** [reply]
If you just changed this

https://github.com/squidfunk/mkdocs-material-insiders/blob/53ac0451e06ff3dd54460682a7143e22fb87769e/src/partials/integrations/analytics/google.html#L115-L116

to 

```js
// Show feedback
if (!feedback.hide)
    feedback.hidden = false;
```

... then this could be achieved with a oneliner in each MD file:

```js
<script>document.addEventListener("DOMContentLoaded", function() { document.feedback.hide = true })</script>
```

Not the most elegant solution, but maybe that's enough for those who need it.

**Comment by michael-nok** [comment]
I'm really glad you added this feature as I was looking into this exact type of functionality!
Is it possible to use **repo_url** variable as part of the HREF link rather than manually specifying it?

Also, depending on which instance of gitlab you're using, the format might be **/new?issue[title]=[Feedback]+{}**

**Reply by squidfunk** [reply]
Currently not. The link is just plain text (or HTML) that is sourced from `mkdocs.yml`, but maybe you can somehow parametrize `mkdocs.yml`. Maybe by using the `macros` plugin? The link should be kept general, because linking to GitHub is only one option. For example, I'm linking to a custom Google Form on the Material documentation.

**Comment by martinbira** [comment]
Have to say I'm really excited about this feature. Now I only need to figure out how to send form data with page info from our S3 hosted site :) Worst case, an e-mail...

**Reply by squidfunk** [reply]
Why do you need to send form data? It's probably easiest to use an external form, e.g. Google Forms.

**Reply by martinbira** [reply]
I'm just scouting options if there are any objections from our IT department, but yeah I was looking at primarily Google Forms.

**Reply by wilhelmer** [reply]
[Formspree](https://formspree.io/) is an excellent choice for static sites, but not cheap if you want spam protection (and you'll want that).

**Comment by michael-nok** [comment]
I'm not sure if this is a bug, but it doesn't look to work when using Google Analytics 4.

The file **partials/integrations/analytics/google.html** only touches the feedback form if the analytics code doesn't start with "G-".

**Reply by squidfunk** [reply]
See #3170.

**Comment by Stephen-Gates** [comment]
@squidfunk  How do to pass the page url you're providing feedback on to the google form? I can see you've done something tricky on https://github.com/squidfunk/mkdocs-material-insiders/blob/master/mkdocs.yml#L119  but I'm not technical enough to work out how you determined to add `?usp=pp_url&entry.853143515={}` to the url Google Forms provides. 

Perhaps the answer could be added to the documentation?  

Thanks in advance 😄

**Reply by squidfunk** [reply]
You need to get a [prefilled link](https://support.google.com/a/users/answer/9308781?hl=en) for your form and replace the URL field with the `{}` placeholder. Yes, we could divide the section on GitHub issues with tabs into guides with multiple providers. If you can make it work, could you submit a PR?
